internal/xds/translator/translator.go

```diff
@@ -113,10 +136,20 @@ func (t *Translator) processHTTPListenerXdsTranslation(tCtx *types.ResourceVersi
 			protocol = HTTP2
 		}
 
+		// Check if an extension is loaded that wants to modify xDS Routes after they have been generated
 		for _, httpRoute := range httpListener.Routes {
 			// 1:1 between IR HTTPRoute and xDS config.route.v3.Route
 			xdsRoute := buildXdsRoute(httpRoute, xdsListener)
-			vHost.Routes = append(vHost.Routes, xdsRoute)
+
```

Comment:
can we achieve this without affecting the core translator logic ?

Reply:
As much of it as possible has been pulled out of the core translator logic without entirely re-processing things "post-translator.go"